diff -ru a/src/gcc/gcc/config/arm/arm-c.c b/src/gcc/gcc/config/arm/arm-c.c
--- a/src/gcc/gcc/config/arm/arm-c.c	2016-02-24 04:19:12.000000000 -0400
+++ b/src/gcc/gcc/config/arm/arm-c.c	2016-08-13 16:55:03.509143808 -0300
@@ -189,6 +189,8 @@
   builtin_assert ("cpu=arm");
   builtin_assert ("machine=arm");
 
+  builtin_define ("__vita__");
+
   arm_cpu_builtins (pfile);
 }
 
diff -ru a/src/gcc/gcc/gcc.c b/src/gcc/gcc/gcc.c
--- a/src/gcc/gcc/gcc.c	2016-02-19 18:18:38.000000000 -0400
+++ b/src/gcc/gcc/gcc.c	2016-08-13 17:13:54.104949572 -0300
@@ -658,8 +658,9 @@
 #endif
 
 /* config.h can define LIB_SPEC to override the default libraries.  */
+#undef LIB_SPEC
 #ifndef LIB_SPEC
-#define LIB_SPEC "%{!shared:%{g*:-lg} %{!p:%{!pg:-lc}}%{p:-lc_p}%{pg:-lc_p}}"
+#define LIB_SPEC "%{!shared:%{g*:-lg} %{!p:%{!pg:-lc}}%{p:-lc_p}%{pg:-lc_p}} -lSceRtc_stub -lSceSysmem_stub -lSceKernelThreadMgr_stub -lSceKernelModulemgr_stub -lSceIofilemgr_stub -lSceProcessmgr_stub -lSceLibKernel_stub -lSceNet_stub"
 #endif
 
 /* When using -fsplit-stack we need to wrap pthread_create, in order
 
diff --ru a/src/gcc/libstdc++-v3/src/c++11/random.cc b/src/gcc/libstdc++-v3/src/c++11/random.cc
--- a/src/gcc/libstdc++-v3/src/c++11/random.cc
+++ b/src/gcc/libstdc++-v3/src/c++11/random.cc
@@ -31,6 +31,10 @@
 # include <cpuid.h>
 #endif
 
+#if defined __vita__
+# include <psp2/kernel/rng.h>
+#endif
+
 #include <cerrno>
 #include <cstdio>
 
@@ -95,6 +99,9 @@ namespace std _GLIBCXX_VISIBILITY(default)
 		return;
 	      }
 	  }
+#elif defined __vita__
+	_M_file = nullptr;
+	return;
 #endif
 
 	fname = "/dev/urandom";
@@ -133,6 +140,15 @@ namespace std _GLIBCXX_VISIBILITY(default)
     result_type __ret;
     void* p = &__ret;
     size_t n = sizeof(result_type);
+
+#if defined __vita__
+	if (!_M_file)
+	{
+		sceKernelGetRandomNumber(p, n);
+		return __ret;
+	}
+#endif
+
 #ifdef _GLIBCXX_HAVE_UNISTD_H
     do
       {

diff --ru a/src/gcc/libstdc++-v3/configure b/src/gcc/libstdc++-v3/configure
--- a/src/gcc/libstdc++-v3/configure
+++ b/src/gcc/libstdc++-v3/configure
@@ -28816,6 +28816,10 @@ case "${host}" in
   arm*-*-symbianelf*)
     # This is a freestanding configuration; there is nothing to do here.
     ;;
+  arm-vita-eabi*)
+    $as_echo "#define _GLIBCXX_USE_RANDOM_TR1 1" >>confdefs.h
+
+    ;;

   mips*-sde-elf*)
     # These definitions are for the SDE C library rather than newlib.

diff --ru a/src/gcc/libstdc++-v3/crossconfig.m4 b/src/gcc/libstdc++-v3/crossconfig.m4
--- a/src/gcc/libstdc++-v3/crossconfig.m4
+++ b/src/gcc/libstdc++-v3/crossconfig.m4
@@ -8,6 +8,9 @@ case "${host}" in
   arm*-*-symbianelf*)
     # This is a freestanding configuration; there is nothing to do here.
     ;;
+  arm-vita-eabi*)
+    AC_DEFINE(_GLIBCXX_USE_RANDOM_TR1)
+    ;;

   mips*-sde-elf*)
     # These definitions are for the SDE C library rather than newlib.
