diff --git a/ld/scripttempl/elf.sc b/ld/scripttempl/elf.sc
index 7381ff0..2394a11 100644
--- a/ld/scripttempl/elf.sc
+++ b/ld/scripttempl/elf.sc
@@ -756,6 +756,10 @@ align_data()
   ${CREATE_SHLIB-${CREATE_PIE-${RELOCATING+. = ${DATA_ADDR-${DATA_SEGMENT_ALIGN}};}}}
   ${CREATE_SHLIB+. = ${SHLIB_DATA_ADDR-${DATA_SEGMENT_ALIGN}};}
   ${CREATE_PIE+. = ${SHLIB_DATA_ADDR-${DATA_SEGMENT_ALIGN}};}
+
+  /* Vita load segments require 64 KiB alignment and optional headroom for SCE metadata */
+  . += DEFINED(__sce_headroom) ? __sce_headroom : 0;
+  . = ALIGN(0x10000);
 EOF
 }
 
