diff -Nru a/src/binutils/ld/scripttempl/elf.sc b/src/binutils/ld/scripttempl/elf.sc
--- a/src/binutils/ld/scripttempl/elf.sc
+++ b/src/binutils/ld/scripttempl/elf.sc
@@ -579,7 +579,25 @@
   ${CREATE_SHLIB-${CREATE_PIE-${RELOCATING+. = ${DATA_ADDR-${DATA_SEGMENT_ALIGN}};}}}
   ${CREATE_SHLIB+. = ${SHLIB_DATA_ADDR-${DATA_SEGMENT_ALIGN}};}
   ${CREATE_PIE+. = ${SHLIB_DATA_ADDR-${DATA_SEGMENT_ALIGN}};}
-
+  /* SCE self module info size */
+  . = . + 0x5C;
+  /* SCE self module_proc_param size */
+  . = . + 0x34;
+  /*
+   * SCE self info rsvd size(import table, etc...)
+   * import library      ( 0x24 byte * number of import library)
+   * import library name (0x100 byte * number of import library)
+   * import opcode       ( 0x10 byte * number of import function * number of import library)
+   * import NID          (    4 byte * number of import function * number of import library)
+   * import pointer      (    4 byte * number of import function * number of import library)
+   * export library      ( 0x20 byte * number of export library)
+   * export library name (0x100 byte * number of export library)
+   * export NID          (    4 byte * number of export function * number of export library)
+   * export pointer      (    4 byte * number of export function * number of export library)
+   */
+  . = . + 0x7000;
+  /* Hacky hackity hack. Need this because otherwise data segment is not aligned enough. */
+  . = ALIGN(0x8000);
   /* Exception handling  */
   .eh_frame     ${RELOCATING-0} : ONLY_IF_RW { KEEP (*(.eh_frame)) ${RELOCATING+*(.eh_frame.*)} }
   .gnu_extab    ${RELOCATING-0} : ONLY_IF_RW { *(.gnu_extab) }
