diff -Nru a/src/binutils/ld/configure.tgt b/src/binutils/ld/configure.tgt
--- a/src/binutils/ld/configure.tgt
+++ b/src/binutils/ld/configure.tgt
@@ -148,7 +148,7 @@
 armeb-*-elf | armeb-*-eabi*)
                        targ_emul=armelfb
                        ;;
-arm-dolce-eabi)
+arm-vita-eabi)
                        targ_emul=armelf_psp2
                        ;;
 arm-*-elf | arm*-*-eabi* | arm-*-rtems*)
