diff -urN binutils-2.34/bfd/elf32-arm.c binutils-2.34-m1/bfd/elf32-arm.c
--- binutils-2.34/bfd/elf32-arm.c	2020-01-18 14:55:47.000000000 +0100
+++ binutils-2.34-m1/bfd/elf32-arm.c	2021-06-17 20:31:37.000000000 +0200
@@ -13730,7 +13730,7 @@
 	     but duplicate entries are likely to be much less common.  */
 	  else
 	    unwind_type = 2;
-
+	  elide = 0;	 
 	  if (elide && !bfd_link_relocatable (info))
 	    {
 	      add_unwind_table_edit (&unwind_edit_head, &unwind_edit_tail,
