diff -Nru a/src/binutils/ld/emulparams/armelf.sh b/src/binutils/ld/emulparams/armelf.sh
--- a/src/binutils/ld/emulparams/armelf.sh
+++ b/src/binutils/ld/emulparams/armelf.sh
@@ -3,7 +3,7 @@
 OUTPUT_FORMAT="elf32-littlearm"
 BIG_OUTPUT_FORMAT="elf32-bigarm"
 LITTLE_OUTPUT_FORMAT="elf32-littlearm"
-TEXT_START_ADDR=0x8000
+TEXT_START_ADDR=0x81000000
 TEMPLATE_NAME=elf
 EXTRA_EM_FILE=armelf
 OTHER_TEXT_SECTIONS='*(.glue_7t) *(.glue_7) *(.vfp11_veneer) *(.v4_bx)'
