diff --git a/gcc/config/arm/arm-c.c b/gcc/config/arm/arm-c.c
index 6e256ee0a..54e390c38 100644
--- a/gcc/config/arm/arm-c.c
+++ b/gcc/config/arm/arm-c.c
@@ -230,6 +230,8 @@ arm_cpu_cpp_builtins (struct cpp_reader * pfile)
   builtin_assert ("cpu=arm");
   builtin_assert ("machine=arm");
 
+  builtin_define ("__vita__");
+
   arm_cpu_builtins (pfile);
 }
 
diff --git a/gcc/config/arm/arm.h b/gcc/config/arm/arm.h
index 9ee6a4eb5..b89f85655 100644
--- a/gcc/config/arm/arm.h
+++ b/gcc/config/arm/arm.h
@@ -669,6 +670,10 @@ extern int arm_arch_cmse;
 #define WCHAR_TYPE_SIZE BITS_PER_WORD
 #endif
 
+/* use int and unsigned int for int32_t and uint32_t */
+#undef STDINT_LONG32
+#define STDINT_LONG32 0
+
 /* Sized for fixed-point types.  */
 
 #define SHORT_FRACT_TYPE_SIZE 8
diff --git a/gcc/gcc.c b/gcc/gcc.c
index 4f57765b0..a4d5ffb14 100644
--- a/gcc/gcc.c
+++ b/gcc/gcc.c
@@ -674,8 +674,9 @@ proper position among the other output files.  */
 #endif
 
 /* config.h can define LIB_SPEC to override the default libraries.  */
+#undef LIB_SPEC
 #ifndef LIB_SPEC
-#define LIB_SPEC "%{!shared:%{g*:-lg} %{!p:%{!pg:-lc}}%{p:-lc_p}%{pg:-lc_p}}"
+#define LIB_SPEC "%{!shared:%{g*:-lg} %{!p:%{!pg:-lc}}%{p:-lc_p}%{pg:-lc_p}} %{pthread:--whole-archive -lpthread --no-whole-archive} -lSceRtc_stub -lSceSysmem_stub -lSceKernelThreadMgr_stub -lSceKernelModulemgr_stub -lSceIofilemgr_stub -lSceProcessmgr_stub -lSceLibKernel_stub -lSceNet_stub"
 #endif
 
 /* When using -fsplit-stack we need to wrap pthread_create, in order
diff --git a/gcc/config/arm/arm.opt b/gcc/config/arm/arm.opt
index af478a946b2..31d0ff7fd18 100644
--- a/gcc/config/arm/arm.opt
+++ b/gcc/config/arm/arm.opt
@@ -21,6 +21,9 @@
 HeaderInclude
 config/arm/arm-opts.h
 
+pthread
+Driver
+
 Enum
 Name(tls_type) Type(enum arm_tls_type)
 TLS dialect to use:
