diff -uHNr gcc-9.1.0.orig/libgcc/gthr-posix.h gcc-9.1.0/libgcc/gthr-posix.h
--- gcc-9.1.0.orig/libgcc/gthr-posix.h	2020-04-16 04:40:41.864017273 +0900
+++ gcc-9.1.0/libgcc/gthr-posix.h	2020-05-01 06:31:36.844109263 +0900
@@ -29,6 +29,9 @@
 /* POSIX threads specific definitions.
    Easy, since the interface is just one-to-one mapping.  */
 
+#undef GTHREAD_USE_WEAK
+#define GTHREAD_USE_WEAK 0
+
 #define __GTHREADS 1
 #define __GTHREADS_CXX0X 1
 
