diff --git a/libgomp/configure b/libgomp/configure
index 5240f7e9d..de5dc96e4 100755
--- a/libgomp/configure
+++ b/libgomp/configure
@@ -15768,29 +15768,6 @@ $as_echo "#define HAVE_UNAME 1" >>confdefs.h
 fi
 rm -f core conftest.err conftest.$ac_objext conftest.$ac_ext
 
-# Check for gethostname.
-cat confdefs.h - <<_ACEOF >conftest.$ac_ext
-/* end confdefs.h.  */
-#include <unistd.h>
-int
-main ()
-{
-
-   char buf[256];
-   if (gethostname (buf, sizeof (buf) - 1) == 0)
-     buf[255] = '\0';
-
-  ;
-  return 0;
-}
-_ACEOF
-if ac_fn_c_try_compile "$LINENO"; then :
-
-$as_echo "#define HAVE_GETHOSTNAME 1" >>confdefs.h
-
-fi
-rm -f core conftest.err conftest.$ac_objext conftest.$ac_ext
-
 # Check for getpid.
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
diff --git a/libgomp/omp.h.in b/libgomp/omp.h.in
--- a/libgomp/omp.h.in
+++ b/libgomp/omp.h.in
@@ -148,7 +148,8 @@
   omp_atk_fallback = 5,
   omp_atk_fb_data = 6,
   omp_atk_pinned = 7,
-  omp_atk_partition = 8
+  omp_atk_partition = 8,
+  __omp_alloctrait_key_t_max__ = __UINTPTR_MAX__
 } omp_alloctrait_key_t;
 
 typedef enum omp_alloctrait_value_t
